subsys/bluetooth/common/bt_to_str.c

```diff
@@ -0,0 +1,63 @@
+/* Copyright (c) 2022 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* Helper for printk parameters to convert from binary to hex.
+ * We declare multiple buffers so the helper can be used multiple times
+ * in a single printk call.
+ */
+
+#include <stddef.h>
+
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/hci.h>
+#include <zephyr/bluetooth/uuid.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/types.h>
+
+const char *bt_hex_real(const void *buf, size_t len)
```

Comment:
Do we still need this function at all? Don't we have this in the logging subsystem now?

Reply:
I guess you are referring to `LOG_HEXDUMP_DBG`? 

I think the two are slightly different, as `LOG_HEXDUMP_DBG` doesn't support formatting, so if you want to log a value and an array, you can't with `LOG_HEXDUMP_DBG`, but you can do

```c
BT_DBG("val %u array val %s", val, bt_hex_real(array_val, sizeof(array_val)));
```

So IMO `LOG_HEXDUMP_DBG` doesn't replace `bt_hex_real`.

Reply:
That makes sense, thanks for the explanation @Thalley. We probably want to move this function to `subsys/logging` though. But we can leave that for later, for now no objections to keeping it here.